pkg/operator/clustermemberremovalcontroller/clustermemberremovalcontroller.go

```diff
@@ -113,19 +113,22 @@ func (c *clusterMemberRemovalController) sync(ctx context.Context, syncCtx facto
 // attemptToScaleDown attempts to remove a voting member only once we have identified that
 // a Machine resource is being deleted and a replacement member has been created
 func (c *clusterMemberRemovalController) attemptToScaleDown(ctx context.Context, recorder events.Recorder) error {
+	klog.V(2).Infof("@Mustafa: enter attemptToScaleDown()")
 	currentVotingMemberIPListSet, err := ceohelpers.VotingMemberIPListSet(c.configMapListerForTargetNamespace)
 	if err != nil {
 		return err
 	}
-
+	klog.V(2).Infof("@Mustafa: currentVotingMemberIPListSet is %v", currentVotingMemberIPListSet)
 	desiredControlPlaneReplicasCount, err := ceohelpers.ReadDesiredControlPlaneReplicasCount(c.operatorClient)
 	if err != nil {
 		return err
 	}
+	klog.V(2).Infof("@Mustafa: desiredControlPlaneReplicasCount is %v", desiredControlPlaneReplicasCount)
 	if desiredControlPlaneReplicasCount == 0 {
 		return fmt.Errorf("desired control plane replicas count cannot be empty")
 	}
-	if currentVotingMemberIPListSet.Len() <= desiredControlPlaneReplicasCount {
+	klog.V(2).Infof("@Mustafa: currentVotingMemberIPListSet.Len() is %v", currentVotingMemberIPListSet.Len())
+	if currentVotingMemberIPListSet.Len() < desiredControlPlaneReplicasCount {
```

Comment:
```suggestion
	if liveVotingMembers, err := c.getAllVotingMembers(ctx); err == nil && len(liveVotingMembers) < desiredControlPlaneReplicasCount {
```

@hasbro17  I propose this change, since the etcd-config map takes some time to remove member's IP after being removed from the membership 

@tjungblu  wdyt ? 
cc @JoelSpeed

Reply:
Just so I understand the change, is the change to make sure we are getting the current voting members from the etcd cluster itself rather than from the configmap? If that's the case that seems like a sensible move to make sure we are not going to break quorum at all, so +1 from me

Reply:
@JoelSpeed  yes exactly, this is the reason .. also all test passed :D 🎉 

i am testing now and will update with the result, before cleaning the logs and adding tests